packages/workflow-runtime/internal/interceptors.ts

```diff
@@ -0,0 +1,136 @@
+  constructor(info: WorkflowInfo) {
+    this.info = info;
+  }
+
+  async scheduleActivity(
+    input: ActivityInput,
+    next: Next<WorkflowOutboundCallsInterceptor, "scheduleActivity">
+  ): Promise<unknown> {
+    const activityType = input.activityType;
+
+    workflowLog(this.info, `Scheduling activity ${activityType}: ${logStringify(input)}`);
+    try {
+      const result = await next(input);
+      workflowLog(this.info, `Activity ${activityType} result: ${logStringify(result)}`);
+      return result;
+    } catch (error) {
+      workflowLog(this.info, `Activity ${activityType} errored: ${error}`);
+      throw error;
+    }
+  }
+
+  async scheduleLocalActivity(
+    input: LocalActivityInput,
+    next: Next<WorkflowOutboundCallsInterceptor, "scheduleLocalActivity">
+  ): Promise<unknown> {
+    workflowLog(this.info, `Scheduling local activity: ${logStringify(input)}`);
+    try {
+      const result = await next(input);
+      workflowLog(this.info, `Local activity result: ${logStringify(result)}`);
+      return result;
+    } catch (error) {
+      workflowLog(this.info, `Local activity errored: ${error}`);
+      throw error;
+    }
+  }
+
+  async startTimer(
+    input: TimerInput,
+    next: Next<WorkflowOutboundCallsInterceptor, "startTimer">
+  ): Promise<void> {
+    workflowLog(this.info, `Starting timer: ${logStringify(input)}`);
+    try {
+      const result = await next(input);
+      workflowLog(this.info, `Starting timer result: ${logStringify(result)}`);
+      return result;
+    } catch (error) {
+      workflowLog(this.info, `Timer errored: ${error}`);
+      throw error;
+    }
+  }
+}
+
+// Interceptor that allows us to log inbound client calls made to workflow.
+// See https://docs.temporal.io/docs/typescript/interceptors for details.
+class WorkflowLogInboundInterceptor implements WorkflowInboundCallsInterceptor {
+  info;
+  constructor(info: WorkflowInfo) {
+    this.info = info;
+  }
+
+  async execute(
+    input: WorkflowExecuteInput,
+    next: Next<WorkflowInboundCallsInterceptor, "execute">
+  ): Promise<unknown> {
+    workflowLog(this.info, `Workflow execution starting: ${logStringify(input)}`);
+    try {
+      const result = await next(input);
+      workflowLog(this.info, `Workflow execution result: ${logStringify(result)}`);
+      return result;
+    } catch (error) {
+      workflowLog(this.info, `Workflow execution errored: ${error}`);
+      throw error;
+    }
+  }
+
+  async handleSignal(
+    input: SignalInput,
+    next: Next<WorkflowInboundCallsInterceptor, "handleSignal">
+  ): Promise<void> {
+    workflowLog(this.info, `Handling signal: ${logStringify(input)}`);
+    try {
+      const result = await next(input);
+      workflowLog(this.info, `Signal result: ${logStringify(result)}`);
+      return result;
+    } catch (error) {
+      workflowLog(this.info, `Error handling signal: ${error}`);
+      throw error;
+    }
+  }
+}
+
+export const workflowInterceptors = () => ({
```

Comment:
nit: since it's already in a package called `workflow-runtime`

```suggestion
export const interceptors = () => ({
```

Reply:
Fixed.